src/language/records.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Records
+description: Summary of the record data structure in Dart.
+toc: false
+---
+
+Records are an anonymous, immutable, aggregate type. Like other [collection types][], 
+they let you bundle multiple objects into a single object. Unlike other collection 
+types, records are fixed-sized, heterogeneous, and typed.
+
+Records are real values; you can store them in variables, pass them to and from
+functions, and store them in lists.
+
+## Record syntax
+
+Records are comma-delimited lists of named or positional fields,
+enclosed in parenthesis:
+
+```dart
+var record = ('first', a: 2, b: true, 'last');    // Record expression
+```
+
+Named fields inside record type annotations go inside a brace-delimited section
+of type and name pairs:
+
+```dart
+(Sring, {int a, bool b}, String?) record;    // Record type annotation
+```
+Naming fields in record type annotations is optional for documentation purposes.
+The names don't need to match any record expression corresponding to the type.
+
+## Record fields
+
+Record fields are accessible through built-in getters. Records are immutable,
+so fields do not have setters. 
+
+Named fields expose getters of the same name. Positional fields expose getters
+of the name `$<position>`, skipping named fields:
+
+```dart
+print(record.$1);    // Prints '123'
```

Comment:
Might be worth repeating the declaration of `record` here in case it's not clear that it's coming from the previous example.